### ISSUE TITLE
Tax forms: add job to cleanup deprecated requests

### ISSUE THREAD

**Issue by Betree** [opencollective/opencollective#3691]
**Describe the bug**
[This](https://opencollective.com/pixijs/expenses/27964) expense was initially submitted with an amount of $1000 then updated to $75 before being approved. Because we send the tax form when the expense is `PENDING` (so right after its creation), user got a tax form request.

Another problem with the current behavior is that a spammer submitting many invoices could exhaust our HelloWorks usage limits.

**Expected behavior**
Only consider expenses for tax form after they've been approved.

**Comment by Betree** [comment]
As discussed in the call today, we don't want to delay sending the tax form email because it could impact the completion negatively because users will not have the context of the expense if they receive it a few days/weeks after.

We'll rather solve this issue by creating a (daily, weekly?) job to cleanup unnecessary tax forms.

**Comment by alanna** [comment]
Yes - we designed it this way because it can create a big delay if a user submits and expense, then has to wait for approval, then has to get and fill out the tax form before getting paid.

**Comment by alanna** [comment]
@Betree is this still an issue?

**Comment by Betree** [comment]
Yes

**Comment by Betree** [comment]
Not needed anymore as we've internalized the tax form system.
